### PR TITLE
Add mirage-types and bump x509 version in ocamlPackages

### PR DESCRIPTION
**Pull request by sternenseemann**
I would be happy to receive feedback on how to improve mirage-types.

(Sorry, I missed the commit msg guideline :/ )

### CODE REVIEW

File: pkgs/development/ocaml-modules/mirage-types/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+stdenv.mkDerivation {
+  name = "ocaml-mirage-types-${version}";
+
+  src  = fetchurl {
+    url = "https://github.com/mirage/mirage/archive/v${version}.zip";
+    sha256 = "bde05f1321e016c09e78db8f2b18781243b6e71cbb81706d15413a432ee597e4";
+  };
+
+  builder = ./build.sh;
```
Comment (jagajaga) [inline pkgs/development/ocaml-modules/mirage-types/default.nix]: Why this old style?

**Comment by jagajaga** [comment]
Please properly name commit's to follow our `CONTRIBUTING.md`. And also please squash commits into one.

**Comment by vcunat** [comment]
> And also please squash commits into one.

Here I think that updating one package and adding another one can well be separate commits.

**Comment by jagajaga** [comment]
@vcunat you are right.

File: pkgs/development/ocaml-modules/mirage-types/default.nix (hunk 1 above)
Comment (lukasepple) [inline pkgs/development/ocaml-modules/mirage-types/default.nix]: Good point, will fix.

**Comment by jagajaga** [comment]
Why did you close?

**Comment by lukasepple** [comment]
I opened an individual ocaml-x509 PR and will work on the rest.

**Pull request closed without merging**